### PR TITLE
Version 1.76.0

### PR DESCRIPTION
**Pull request by SC426**
rust-activation 1.76.0

**Destination channel:** defaults

### Links

- [PKG-4239](https://anaconda.atlassian.net/browse/PKG-4239)
- [Upstream repository](https://github.com/rust-lang/rust/tree/1.76.0)
- [Upstream changelog/diff](https://github.com/rust-lang/rust/releases)

### Explanation of changes:

- Update version and `sha256`
- Linter fixes
- Update tests
- Change clang deployment target version for `osx-64` to [10.12](https://github.com/rust-lang/rust/pull/104385)

[PKG-4239]: https://anaconda.atlassian.net/browse/PKG-4239?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by skupr-anaconda — COMMENTED** [review]
It's not typical for `m2w64-toolchain  # [win]` to be in the `run` section. Because it's a build tool. It's dependencies are:
`m2w64-binutils, m2w64-crt-git, m2w64-gcc, m2w64-gcc-ada, m2w64-gcc-fortran, m2w64-gcc-libgfortran, m2w64-gcc-libs, m2w64-gcc-objc, m2w64-headers-git, m2w64-libmangle-git, m2w64-libwinpthread-git, m2w64-make, m2w64-pkg-config, m2w64-tools-git, m2w64-winpthreads-git, msys2-conda-epoch >=20160418`

But I'm not familiar with activation scripts

**Comment by danpetry** [comment]
I'm not either, but I think it means that if the activation packages are used in a build environment, they pull in their compilers and other required tools to the build environment too. In other words, the activation packages need the toolchain to run (in a build environment)

**Review by danpetry — APPROVED** [review]

**Comment by ryanskeith** [comment]
I know it may be bothersome but there is a fundamental difference in how we do rust activations versus conda forge. On everything but windows, we create a file but we should just make use of environment variables.

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,6 +13,10 @@ outputs:
     script: bld-msvc.bat  # [win]
     build:  # [win and int(float(vc)) != 14]
       skip: True  # [win and int(float(vc)) != 14]
+    build:
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:16]: Should these sections be merged?

Comment (SC426) [reply recipe/meta.yaml]: Done. Thank you!

**Pull request merged**